backend-kotlin/src/test/kotlin/com/example/backendkotlin/usecase/CreateVillageUseCaseUT.kt

```diff
@@ -54,29 +72,95 @@ class CreateVillageUseCaseUT(
                     unmockkStatic(BCrypt::class)
+
+                    verify(exactly = 1) { userRepository.createUser(gameMaster) }
+                    verify(exactly = 1) { villageRepository.createVillage(village, hashedPassword, salt) }
+                    verify(exactly = 1) { rUserVillageRepository.save(gameMaster.id, village.id) }
                 }
             }
             context("異常系") {
+                it("ゲームマスターが作成されたが、IDが一致しない") {
+                    // given
+                    val requestedGameMaster = User(
+                        id = UserId(UUID.randomUUID()),
+                        name = "gameMaster",
+                        isActive = true,
+                    )
+                    val village = Village(
+                        id = VillageId(UUID.randomUUID()),
+                        name = "村1",
+                        citizenCount = 10,
+                        werewolfCount = 2,
+                        fortuneTellerCount = 1,
+                        knightCount = 1,
+                        psychicCount = 1,
+                        madmanCount = 1,
+                        isInitialActionActive = true,
+                        gameMasterUserId = requestedGameMaster.id,
+                    )
+                    val password = "password"
+                    val salt = "salt"
+                    val hashedPassword = "hashedPassword"
+
+                    // and
+                    // idの異なるゲームマスターが作成されたとする
+                    val createdGameMaster = User(
+                        id = UserId(UUID.randomUUID()),
+                        name = "gameMaster",
+                        isActive = true,
+                    )
+                    every { userRepository.createUser(requestedGameMaster) } returns createdGameMaster
+                    mockkStatic(BCrypt::class)
+                    every { BCrypt.gensalt() } returns salt
+                    every { BCrypt.hashpw(password, salt) } returns hashedPassword
+                    every { BCrypt.checkpw(password, hashedPassword) } returns true
+                    every { villageRepository.createVillage(village, hashedPassword, salt) } returns village
+                    every { rUserVillageRepository.save(createdGameMaster.id, village.id) } returns Pair(createdGameMaster.id, village.id)
+
+                    // when, then
+                    val exception = shouldThrowExactly<RuntimeException> {
+                        target.invoke(village, password, requestedGameMaster)
+                    }
+
+                    // then
+                    exception.message shouldBe "Game master has created, but id is not matched"
+                    verify(exactly = 1) { userRepository.createUser(requestedGameMaster) }
+                    verify(exactly = 0) { villageRepository.createVillage(village, hashedPassword, salt) }
+                    verify(exactly = 0) { rUserVillageRepository.save(createdGameMaster.id, village.id) }
```

Comment:
[FYI] 地味嬉しいポイントなんですが、 exactly が同じ部分は 1 block にまとめられます
```suggestion
                    verify(exactly = 0) { 
                        villageRepository.createVillage(village, hashedPassword, salt)
                        verify(exactly = 0) { rUserVillageRepository.save(createdGameMaster.id, village.id) 
                    }
```

Reply:
19c48cc で修正